### ISSUE TITLE
Expose swarm hostname

### ISSUE THREAD

**Issue by ainsey11** [joonas-fi/promswarmconnect#4]
Love this so far, however is it possible to differentiate which node the metrics instance is running on?
a prime example of this would be cadvisor, 

say I have a three node cluster, I end up with three discovered cadvisor containers, something a little like 
![image](https://user-images.githubusercontent.com/6608275/53613295-94113180-3bcd-11e9-98d9-24f8afcfb6ef.png)

Is there a way I can tell prometheus which node each of these lives on? the only way I can differentate between the three is the vm_alias field which returns the containers IP address, this would be very useful with things like grafana etc,

Thanks!
Ainsey

**Comment by joonas-fi** [comment]
Glad you found this useful!

If I understood correctly, yes, I'm also doing that because I have node-specific metrics I want to observe in production.

If you have a per Swarm-node service and you want to have it show up in Prometheus by the server hostname as it's `instance` value, you can override it like this: https://github.com/function61/promswarmconnect/blob/7b450c4767c12f001b0d0318bf4bff68fffbfaa7/cmd/promswarmconnect/metricsendpointparser_test.go#L68

Basically, when creating Swarm service add an ENV var: `$ docker service create -e METRICS_ENDPOINT=/metrics,instance=_HOSTNAME_`

The `_HOSTNAME_` is a magical placeholder which promswarmconnect will replace at discovery time.

**Comment by ainsey11** [comment]
Thanks!
That worked perfectly, it now returns "vm_uuid": "nc-docker-1"  etc,

**Comment by joonas-fi** [comment]
Great!

I need to work on the docs to explain these better. This override was
linked to from the README but it's not clear enough, and it even refers to
a deprecated way of doing it (different ENV variable, which I would like to
remove)

Have a nice weekend!

**Comment by ainsey11** [comment]
No worries,

I was in the progress of tweaking the readme to make it a little clearer, hopefully that helps!

- Ainsey
